src/streamlink/plugins/youtube.py

```diff
@@ -214,84 +209,27 @@ def _create_adaptive_streams(self, info, streams, protected):
 
         return streams, protected
 
-    def _find_channel_video(self):
-        res = http.get(self.url)
+    def _find_video_id(self, url):
+        res = http.get(url)
+
+        for link in itertags(res.text, 'link'):
+            if link.attributes.get("rel") == "canonical":
```

Comment:
This will redirect `/user/` to `/channel/` URL, but it's the same page so you will open the same page twice.

https://www.youtube.com/user/france24
https://www.youtube.com/channel/UCCCPCZNChQdGa9EkATeye4g

---

it should be ignored for `/live` urls, they have a videoID somewhere on the page

Reply:
I switched it so it only goes to the canonical page if the videoID cannot be found using the other methods.